R/param-estimates-batch.R

```diff
@@ -103,3 +106,101 @@ param_estimates_batch <- function(.path,
       everything()
     )
 }
+
+
+#' Compare parameter estimates
+#'
+#' Summarizes the parameter estimates from tibble like what's returned from
+#' [param_estimates_batch()], showing you the quantiles passed to `probs`,
+#' likely interpreted as confidence intervals for your parameter estimates.
+#' Optionally takes an "original model" to compare these quantiles against.
+#' Originally conceived for comparing a model to multiple runs of the "same"
+#' model, for example in a bootstrap or simulation.
+#'
+#' @return A tibble with the first column containing the parameter names (that
+#'   were column names in `.param_df`), the second column optionally containing
+#'   original parameter estimates (if `.orig_mod` is passed), and subsequent
+#'   columns containing the quantiles specified in `probs`. You can think of
+#'   this as essentially `.param_df`, summarized by quantiles across all rows,
+#'   and then pivoted.
+#'
+#'
+#' @param .param_df A tibble with columns for each parameter and rows for each
+#'   model (like what's returned from [param_estimates_batch()])
+#' @param .orig_mod `bbi_model` object to compare `.param_df` against. If
+#'   `NULL`, the default, only returns quantiles from `.param_df`. If passed,
+#'   will display an additional `original_estimate` column.
+#' @param .compare_cols An expression that can be passed to [dplyr::select()] to
+#'   select which columns in `.param_df` will be pivoted and summarized.
+#'   Defaults to `dplyr::starts_with(c("THETA", "SIGMA", "OMEGA"))` (designed
+#'   for NONMEM runs), but consider using [dplyr::matches()] to select columns
+#'   using regex. Also see
+#'   [?tidyselect::language](https://tidyselect.r-lib.org/reference/language.html)
+#'    for more details and options.
+#' @param probs Numeric vector with values between 0 and 1 to be passed through to
+#'   [stats::quantile()]. Represents the quantiles to calculate for parameter
+#'   estimates in `.param_df`.
+#' @param na.rm Logical scalar, passed through to [stats::quantile()].
+#'
+#' @importFrom tibble rownames_to_column
+#' @importFrom dplyr summarise across starts_with select rename left_join
+#' @importFrom stats quantile
+#' @export
+param_estimates_compare <- function(
+  .param_df,
+  .orig_mod = NULL,
+  .compare_cols = starts_with(c("THETA", "SIGMA", "OMEGA")),
+  probs = c(.5, 0.025, 0.975),
+  na.rm = FALSE
+) {
+
+  comp_df <- .param_df %>%
+    select(.compare_cols)
```

Comment:
I think you'll need to use `{{ .compare_cols }}` to get the expected evaluation.

https://dplyr.tidyverse.org/articles/programming.html#indirection-3

```
> param_estimates_compare(param_df, .compare_cols = THETA2)
Error in `instrument_base_errors()`:
! object 'THETA2' not found
Caused by error in `env_get()`:
! object 'THETA2' not found
Run `rlang::last_error()` to see where the error occurred.

# with `{{ .compare_cols }}`
> devtools::load_all()
ℹ Loading bbr
setting `bbr.DEV_no_min_version` to TRUE so no validation of bbi version will occur
> param_estimates_compare(param_df, .compare_cols = THETA2)
  parameter_names     50%    2.5%   97.5%
1          THETA2 54.6151 54.6151 54.6151
```

Reply:
> In my view, the expected evaluation for the tidy selection should be used. Is there some downside of using {{ .compare_cols }} that I'm overlooking?

No, I guess there isn't. And considering these other cases, I think you're right we should switch to embrace. Would you change the docs for the argument (from what they are in the PR now) or do you think it's accurate and clear as is?

Reply:
The description seems clear to me as is.